exec/internal/compile/optimizer_amd64.go

```diff
@@ -0,0 +1,35 @@
+package compile
+
+import (
+	"math"
+
+	asm "github.com/twitchyliquid64/golang-asm"
+	"github.com/twitchyliquid64/golang-asm/obj"
+	"github.com/twitchyliquid64/golang-asm/obj/x86"
+)
+
+func peepholeOptimizeAMD64(builder *asm.Builder) error {
```

Comment:
would you have a simple idea on how to directly test this function?
would it be relatively easy?

Reply:
Its possible, but it wouldnt be easy as instructions are stored as a linked list and id have to build the input list and then test the output one via some messy iterator.

Reply:
ok